src/main/java/picard/illumina/CheckIlluminaDirectory.java

```diff
@@ -168,9 +168,8 @@ protected int doWork() {
                     filterFileMap.put(fileToTile(filterFile.getName()), filterFile);
                 }
 
-                final OutputMapping outputMapping = new OutputMapping(readStructure);
-
-                final CbclReader reader = new CbclReader(cbcls, filterFileMap, readStructure.readLengths, tiles.get(0), locs, outputMapping.getOutputCycles(), true);
+                final CbclReader reader = new CbclReader(cbcls, filterFileMap, outputMapping.getOutputReadLengths(),
```

Comment:
It looks like the significant change here is replacing `readStructure.readLengths` with `outputMapping.getOutputReadLengths()`. Why did you remove the variable `expectedCycles` which was caching `outputMapping.getOutputCycles()` as well?

Reply:
I'm caching the full outputmapping now. It is one extra get call now. Not sure its worth caching that as well.

Reply:
Well, it's not a simple getter (it calls `ReadStructure.getCycles()`), but I'll leave it up to you if it's worth caching or not.